packages/nimble-components/src/wafer-map/index.ts

```diff
@@ -131,18 +128,43 @@ export class WaferMap extends FoundationElement {
             this.dieLabelsSuffix,
             this.maxCharacters
         );
-        this.renderer = new RenderingModule(this.dataManager, this.canvas);
-        this.zoomHandler = new ZoomHandler(
-            this.canvas,
-            this.zoomContainer,
-            this.dataManager,
-            this.renderer,
-            this.canvasSideLength
+
+        const renderer = new RenderingModule(dataManager, this.canvas);
+        this.eventCoordinator = new EventCoordinator(
+            this.parseWaferDataToEventData(dataManager, renderer, this)
         );
-        this.zoomHandler.attachZoomBehavior();
-        this.renderer.drawWafer();
+        renderer.drawWafer();
     }
 
+    private createResizeObserver(): ResizeObserver {
+        const resizeObserver = new ResizeObserver(entries => {
+            const entry = entries[0];
+            if (entry === undefined) {
+                return;
+            }
+            const { height, width } = entry.contentRect;
+            this.canvasSideLength = Math.min(height, width);
+        });
+        resizeObserver.observe(this);
+        return resizeObserver;
+    }
+
+    private cleanUp(): void {
+        this.clearCanvas(this.canvasSideLength, this.canvasSideLength);
+        if (this.eventCoordinator !== undefined) {
+            this.eventCoordinator.detachEvents();
+        }
+    }
+
+    private clearCanvas(width: number, height: number): void {
+        const context = this.canvas.getContext('2d')!;
+        context.clearRect(0, 0, width, height);
+    }
+
+    private readonly emitDieSelected = (die: WaferMapDie): void => {
+        this.$emit('die-selected', { detail: { die } });
```

Comment:
We should document this event by adding it to the storybook for the wafer map.

Reply:
applied - as an action